src/main/java/org/jfrog/hudson/release/PromotionUtils.java

```diff
@@ -20,83 +14,46 @@
-            try (CloseableHttpResponse dryResponse = client.stageBuild(buildName, buildNumber, promotion)) {
-                entity = dryResponse.getEntity();
-                validatePromotionSuccessful(dryResponse, true, promotion.isFailFast(), listener);
+
+            try {
+                artifactoryManager.stageBuild(buildName, buildNumber, promotion);
+                listener.getLogger().println("Dry run finished successfully.\nPerforming promotion ...");
             } catch (IOException e) {
-                listener.error(e.getMessage());
-                return false;
-            } finally {
-                EntityUtils.consumeQuietly(entity);
+                if (onPromotionFailFast(true, promotion.isFailFast(), listener)) {
+                    return false;
+                }
+                listener.getLogger().println("Dry run failed to perform promotion ...");
             }
-            listener.getLogger().println("Dry run finished successfully.\nPerforming promotion ...");
         }
 
         // Perform promotion
         promotion.setDryRun(false);
-        try (CloseableHttpResponse response = client.stageBuild(buildName, buildNumber, promotion)) {
-            entity = response.getEntity();
-            validatePromotionSuccessful(response, false, promotion.isFailFast(), listener);
+        try {
+            artifactoryManager.stageBuild(buildName, buildNumber, promotion);
         } catch (IOException e) {
             listener.error(e.getMessage());
             return false;
-        } finally {
-            EntityUtils.consumeQuietly(entity);
         }
         listener.getLogger().println("Promotion completed successfully!");
 
         return true;
     }
 
-    /**
-     * Checks the status of promotion response and return true on success
-     *
-     * @param response
-     * @param dryRun
-     * @param failFast
-     * @param listener
-     * @return
-     */
-    public static void validatePromotionSuccessful(HttpResponse response, boolean dryRun, boolean failFast, TaskListener listener) throws IOException {
-        StatusLine status = response.getStatusLine();
-        String content;
-
-        try {
-            content = ExtractorUtils.entityToString(response.getEntity());
-        } catch (IOException e) {
-            throw new IOException("Failed parsing promotion response.", e);
-        }
-
-        if (failOnResponseStatus(dryRun, failFast, listener, status, content)) {
-            throw new IOException("Promotion failed due to Artifactory response.");
-        }
-
-        listener.getLogger().println(content);
-    }
-
-    private static boolean failOnResponseStatus(boolean dryRun, boolean failFast, TaskListener listener, StatusLine status, String content) {
-        if (status.getStatusCode() != 200 && failFast) {
+    public static boolean onPromotionFailFast(boolean dryRun, boolean failFast, TaskListener listener) {
```

Comment:
Let's add the exception to this method and print it in the listener.error.
Also, I think we don't really need the boolean return value.